core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -828,37 +645,9 @@ private PrivateKey getTenantPrivateKey(String keyStoreName, String alias) throws
         if (log.isDebugEnabled()) {
             log.debug("Loading private key from tenant key store : " + keyStoreName + " alias: " + alias);
         }
-        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
-        Resource resource;
-        KeyStore keyStore;
-
-        if (registry.resourceExists(path)) {
-            resource = registry.get(path);
-        } else {
-            throw new SecurityException("Given Key store is not available in registry : " + keyStoreName);
-        }
-
-        CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
-        String encryptedPassword = resource
-                .getProperty(RegistryResources.SecurityManagement.PROP_PRIVATE_KEY_PASS);
-        String privateKeyPasswd = new String(cryptoUtil.base64DecodeAndDecrypt(encryptedPassword));
-
-        if (isCachedTenantKeyStoreValid(keyStoreName)) {
-            keyStore = tenantKeyStores.get(keyStoreName).getKeyStore();
-        } else {
-            byte[] bytes = (byte[]) resource.getContent();
-            String keyStorePassword = new String(cryptoUtil.base64DecodeAndDecrypt(resource.getProperty(
-                    RegistryResources.SecurityManagement.PROP_PASSWORD)));
-            keyStore = KeystoreUtils.getKeystoreInstance(resource
-                    .getProperty(RegistryResources.SecurityManagement.PROP_TYPE));
-            ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
-            keyStore.load(stream, keyStorePassword.toCharArray());
-
-            KeyStoreBean keyStoreBean = new KeyStoreBean(keyStore, resource.getLastModified());
-            updateTenantKeyStoreCache(keyStoreName, keyStoreBean);
```

Comment:
Why removed cache update?

Reply:
This cache update also done in the getKeyStore() method. https://github.com/wso2/carbon-kernel/pull/4109/files#diff-487286dcf4015c4afee9863f0dfa368ba508c68413e0a8ee9315f93e5cc7d547R469